pkg/storage/couchdb/couchdbstore.go

```diff
@@ -253,20 +253,44 @@ func (c *CouchDBStore) Put(k string, v []byte) error {
 	return nil
 }
 
-// GetAll fetches all the key-value pairs within this store.
-// TODO: #61 Add support for pagination.
-func (c *CouchDBStore) GetAll() (map[string][]byte, error) {
-	rows, err := c.db.AllDocs(context.Background(), kivik.Options{"include_docs": true})
+// PutAll stores the key-value pairs in the order given in the array. The end result is equivalent to calling
+// Put(k,v) on each key-value pair individually in a loop, but should be faster since this method minimizes REST calls.
+func (c *CouchDBStore) PutAll(keys []string, values [][]byte) error {
+	err := validateKeysAndValues(keys, values)
 	if err != nil {
-		return nil, fmt.Errorf(failureWhileGettingAllDocs, err)
+		return err
 	}
 
-	allKeyValuePairs, err := c.getAllKeyValuePairs(rows)
+	// If CouchDB receives the same key multiple times, it will just keep the first change and disregard the rest.
```

Comment:
Is there anything strange sounding about this? Is my conclusion sound?

Reply:
Yes, we do want the last one since you are replacing document. I am wondering if we need to worry about some kind of document history here (dropping versions impacts true history). You could have two or more putAll in case we have duplicates - we can discuss tomorrow if document history (revisions) is important in this case.

Reply:
Good point. Yes, indeed there's a subtle difference between calling my new `PutAll` and calling `Put` in a loop. I've added this point to the method comment and also created a follow up issue to provide an option for preserving history (at the expense of speed to some degree): https://github.com/trustbloc/edge-core/issues/120

Reply:
@sandrask Note that CouchDB doesn't actually do full versioning: https://stackoverflow.com/questions/9730299/are-old-data-accessible-in-couchdb

Reply:
(your point still stands, technically the history from CouchDB's perspective is different)